### DISCUSSION TITLE
Guest, template and external home

### DISCUSSION THREAD

**Discussion by gfreccia** [ltsp/ltsp#178] [Q&A]
Hi Alkis,

I've plans to install new LTSP for my school laboratories soon!

Do you have news about "guest" session or about creation of a template for all user?

I'm planning to use a dedicated server for "home" directory of all my users. Has it sense? Any hints on it?

Then... could you please explain better what this commands mean?

_Virtual machines need to be symlinked before running ltsp image:

ln -s "/home/user/VirtualBox VMs/debian/debian-flat.vmdk" /srv/ltsp/debian.img
ltsp image debian

For a chroot in /srv/ltsp/x86_32:

ltsp image x86_32_

Thank you!

**Comment by alkisg** [comment]
Hi Giacomo,

about the guest account, define what you mean with guest:

- Do you need autologin, i.e. to automatically connect to that user without showing lightdm at all?
- Do you need passwordless login, i.e. to be able to select "guest", press [enter], and login without using a password?
- Are files saved on logout? Are they cleared on login? Are they cleared on reboot?
- Do you need an empty /home/guest, or a template /home/guest which you have customized?

About the virtual machine commands etc, let's first discuss about your needs. The chrootless method is the easiest one, that way you don't need virtual machines or anything. Why don't you want to use that?

About the dedicated /home, why, to split the NFS (=/) and SSHFS (=/home) bandwidth to two servers? Btw, are you planning to use SSHFS, or is NFS3 fine too? (which is faster but insecure)

**Comment by gfreccia** [comment]
Hi Alkis,

thanks for the prompt response!

What I need for my laboratory
- autologin;
- no files, settings saved at logout (e.g an user who changes desktop);
- template configuration for simple creation of users;
- shared folder on the server linked on the desktop of all users, so the
users can save files manually on that folder.

I've plans to install my server on a Proxmox host.
I assume you still suggest to use MATE!

Dedicated server for "home" is to facilitate management, so I can change
LTSP server without affecting home dir. 

---

Giacomo Trovato

Il 07/09/2019 12:13 Alkis Georgopoulos ha scritto:

> Hi Giacomo, 
> 
> about the guest account, define what you mean with guest: 
> 
> * Do you need autologin, i.e. to automatically connect to that user without showing lightdm at all?
> * Do you need passwordless login, i.e. to be able to select "guest", press [enter], and login without using a password?
> * Are files saved on logout? Are they cleared on login? Are they cleared on reboot?
> * Do you need an empty /home/guest, or a template /home/guest which you have customized?
> 
> About the virtual machine commands etc, let's first discuss about your needs. The chrootless method is the easiest one, that way you don't need virtual machines or anything. Why don't you want to use that? 
> 
> About the dedicated /home, why, to split the NFS (=/) and SSHFS (=/home) bandwidth to two servers? Btw, are you planning to use SSHFS, or is NFS3 fine too? (which is faster but insecure) 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q5A6BVRRB6TRNHTMITQIN5F5A5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6EVKLQ#issuecomment-529093934
[2]

**Comment by alkisg** [comment]
OK so I suggest the following:

- Use the chrootless method. No virtual machines involved.
- Let's call the first account with uid=1000 "administrator". Create a second account and name it "guest". You only need one account for all clients. Login as guest on the server and customize everything as you like.
- Follow the installation page and verify that you can boot a single client and login as guest there.

When you've done those, write here so that I mention the next steps.

In the meantime, also decide how you're planning to use the remote home. Do you want to use SSHFS there, which means that you'll need the "guest account" there too, or do you want the easier, faster, but insecure NFS3, that doesn't require a guest account in the separate home server?

**Comment by gfreccia** [comment]
Hi Alkis,

I want to use the same LTSP server for computer room (autologin with
settings restored at logout) and for teachers users (with personal
account).

I've not constrains about security, so the "easier" NFS3 method is ok.

Do you confirm that it's better to use MATE instead of MINT?

Thank you! 

---

Giacomo Trovato

Il 07/09/2019 13:53 Alkis Georgopoulos ha scritto:

> OK so I suggest the following: 
> 
> * Use the chrootless method. No virtual machines involved.
> * Let's call the first account with uid=1000 "administrator". Create a second account and name it "guest". You only need one account for all clients. Login as guest on the server and customize everything as you like.
> * Follow the installation page and verify that you can boot a single client and login as guest there.
> 
> When you've done those, write here so that I mention the next steps. 
> 
> In the meantime, also decide how you're planning to use the remote home. Do you want to use SSHFS there, which means that you'll need the "guest account" there too, or do you want the easier, faster, but insecure NFS3, that doesn't require a guest account in the separate home server? 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7QYI5FB2ONDHXBIBJSDQIOI3VA5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6EXELQ#issuecomment-529101358
[2]

**Comment by alkisg** [comment]
> Do you confirm that it's better to use MATE instead of MINT?

I can't say it with those words.
What I can say is: "Personally I prefer Ubuntu or Debian MATE to Mint, because Mint breaks Debian policy in various places and makes things more unstable / inconsistent for me."

**Comment by gfreccia** [comment]
Hi Alkis,

I've installed new LTSP on Mate 18.04.3 and finally I've my Asus mini PC
(AS PN40) booting over IPXE (no chance to have it working with old
LTSP)!
It's very fast but I've discovered some freezes using Google Maps on
Firefox.
Could be it related with Issue 27?

Then it seems there is no network swap active and I've not found how to
set it on new lts.conf.
Is that option disappeared?

Thank you! 

---

Giacomo Trovato

Il 08/09/2019 18:00 Alkis Georgopoulos ha scritto:

>> Do you confirm that it's better to use MATE instead of MINT?
> 
> I can't say it with those words.
> What I can say is: "Personally I prefer Ubuntu or Debian MATE to Mint, because Mint breaks Debian policy in various places and makes things more unstable / inconsistent for me." 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q7J43LT6467QNUSSHLQIUOSPA5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6FTGEA#issuecomment-529216272
[2]

**Comment by alkisg** [comment]
If it's freezing and coming back after a few seconds, then yes it might be related.
If it's freezing completely and xorg crashes or kernel panics etc etc, then it sounds like a bug in the graphics/opengl drivers; or an issue with RAM.

NBD_SWAP was causing a lot of issues in LTSP5. I want to take time and do a lot of testing before enabling it in the new LTSP. For now, the new LTSP only supports and automatically uses local swap partitions, if the clients have local disks etc.

**Comment by gfreccia** [comment]
Thanks Alkis!

I hope it's related with the issue since it comes back after some
seconds!
To apply the patch I've just to udpate LTSP with:

_apt update_
_apt full-upgrade_
_ltsp -o ipxe_

How must be partitioned/formatted the local disk to have it working as
swap?
I've just to create a swap partition?

---

Giacomo Trovato

Il 16/09/2019 08:59 Alkis Georgopoulos ha scritto:

> If it's freezing and coming back after a few seconds, then yes it might be related.
> If it's freezing completely and xorg crashes or kernel panics etc etc, then it sounds like a bug in the graphics/opengl drivers; or an issue with RAM. 
> 
> NBD_SWAP was causing a lot of issues in LTSP5. I want to take time and do a lot of testing before enabling it in the new LTSP. For now, the new LTSP only supports and automatically uses local swap partitions, if the clients have local disks etc. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q6SZ2KMRUGSTXPRQTLQJ4VFPA5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6YIITY#issuecomment-531661903
[2]

**Comment by alkisg** [comment]
> I've just to create a swap partition?

Exactly, if LTSP finds a local swap partition it'll use it. Anything from 512 MB to 8 GB should be fine for swap. Personally I run `gparted` on LTSP clients and create a swap partition (possibly shrinking the Windows partition if it exists), then I reboot them and it's working out of the box.

**Comment by alkisg** [comment]
Giacomo, after you've completed the updates etc, reboot the client and run this:

```shell
grep nfs /proc/self/mountinfo
```

Paste the output here, it'll show us if you're still affected or not.

**Comment by gfreccia** [comment]
Here: 

_53 23 0:45 / /proc/fs/nfsd rw,relatime shared:32 - nfsd nfsd rw_

---

Giacomo Trovato

Il 16/09/2019 09:11 Alkis Georgopoulos ha scritto:

> Giacomo, after you've completed the updates etc, reboot the client and run this:
> 
> grep nfs /proc/self/mountinfo
> 
> Paste the output here, it'll show us if you're still affected or not. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q7QNIH7VBPYU5ZW42LQJ4WR3A5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6YI4VI#issuecomment-531664469
[2]

**Comment by alkisg** [comment]
Giacomo was this from the client or the server?
If server => run in on the client.
If client => ok you're not affected by NFS home performance.

**Comment by gfreccia** [comment]
Hi Alkis, 

it was from the server. 

On the client I've no answer on such command!

---

Giacomo Trovato

Il 16/09/2019 14:52 Alkis Georgopoulos ha scritto:

> Giacomo was this from the client or the server?
> If server => run in on the client.
> If client => ok you're not affected by NFS home performance. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q5OCRKO5REXG7EN2WTQJ56R7A5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6ZBCRI#issuecomment-531763525
[2]

**Comment by alkisg** [comment]
OK, that means you're using SSHFS on the client and you're not affected by NFS home entries in ltsp.conf.

**Comment by gfreccia** [comment]
Hi Alkis,

but you had written to me this:

_"In the meantime, also decide how you're planning to use the remote
home. Do you want to use SSHFS there, which means that you'll need the
"guest account" there too, or do you want the easier, faster, but
insecure NFS3, that doesn't require a guest account in the separate home
server?"._

So, how I can set NFS3? I thought that on the new LTSP it was set by
default!

---

Giacomo Trovato

Il 16/09/2019 16:19 Alkis Georgopoulos ha scritto:

> OK, that means you're using SSHFS on the client and you're not affected by NFS home entries in ltsp.conf. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7QY55FHOECWWULEXYP3QJ6IXZA5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6ZJWNQ#issuecomment-531798838
[2]

**Comment by alkisg** [comment]
To use NFS home:
1. You run `ltsp nfs --nfs-home=1` to create an appropriate /etc/exports.d/ltsp-nfs.exports file, and
2. You uncomment the following line in ltsp.conf:
[clients]
FSTAB_NFS="server:/home /home nfs defaults,nolock,rsize=32768,wsize=32768 0 0"

Finally, `ltsp initrd` is needed after every change in ltsp.conf.

**Comment by gfreccia** [comment]
Hi Alkis, 

done! 

Now I've this on mountinfo on the client: 

_/proc/fs/nfsd/ rw,relatime shared:31 - nfsd nfsd rw_

---

Giacomo Trovato

Il 16/09/2019 17:47 Alkis Georgopoulos ha scritto:

> To use NFS home: 
> 
> * You run ltsp nfs --nfs-home=1 to create an appropriate /etc/exports.d/ltsp-nfs.exports file, and
> 
> * You uncomment the following line in ltsp.conf:
> [clients]
> FSTAB_NFS="server:/home /home nfs defaults,nolock,rsize=32768,wsize=32768 0 0" 
> 
> Finally, ltsp initrd is needed after every change in ltsp.conf. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7QYSKTNPWU626IOROGLQJ6TAXA5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6ZTB5A#issuecomment-531837172
[2]

**Comment by gfreccia** [comment]
Hi Alkis, 

I've now my MATE LTSP with external home (on Openmediavault NFS share)
working!

Now I need, for all LTSP users, a common shared folder of my NFS server,
linked on the desktop or on Caja.

Do I've just to add another line on lts.conf? 

e.g: 

_FSTAB_1="192.168.0.89:/share /share nfs defaults,nolock 0 0"_

Thank you! 

---

Giacomo Trovato

Il 16/09/2019 09:11 Alkis Georgopoulos ha scritto:

> Giacomo, after you've completed the updates etc, reboot the client and run this:
> 
> grep nfs /proc/self/mountinfo
> 
> Paste the output here, it'll show us if you're still affected or not. 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/ltsp/community/issues/14?email_source=notifications&amp;email_token=AISJ7Q7QNIH7VBPYU5ZW42LQJ4WR3A5CNFSM4IUPRNX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6YI4VI#issuecomment-531664469
[2]

**Comment by alkisg** [comment]
Hi Giacomo,

first of all, when you're replying to github issues via email, delete all the old mail content, as it looks very ugly online, for example see https://github.com/ltsp/community/issues/14#issuecomment-534933241.

In general it's best to visit https://github.com/ltsp/community/issues/14 and reply via your browser.

To add another share to LTSP clients, yes, you'd use something like:

```shell
FSTAB_SHARE="192.168.0.89:/share /share nfs defaults,nolock 0 0"
```

Make sure that you do have a /share directory on the client image.

**Comment by gfreccia** [comment]
Hi Alkis,

I've the "home" working properly on Openmediavault NFS share.
But to manage more simply the access rights of the users, I think would be more simple to put the "home" of the user on lts.conf.
So it would need to pass the user name as variable; e.g:

 _FSTAB_NFS="server:/home/[user] /home/[user] nfs defaults,nolock,rsize=32768,wsize=32768 0 0"_

Is it possible?

Thank you!

**Comment by alkisg** [comment]
Per user NFS mounts are possible, but they're not easy, and they don't really offer anything that normal mounts don't offer already.

I don't think you need to do anything at all about access rights. If you create a server:/home share, and you export it with the "no_root_squash" option, then LTSP will create /home/username automatically at first login, and properly chown it to the user.

I.e. each user will have access to his own /home/user dir, like it does on local /home, without having to do anything at all.

**Comment by gfreccia** [comment]
Hi Alkis,

thanks!
I apologize continuing to bother you, but I'm very enthusitastic about new LTSP, and I'm trying to swap to it in all my installations, so...

Can I set a user as "template" to create all my other users?
Is there a way to disable the user to change settings on desktop (e.g. wallpaper)?

**Comment by alkisg** [comment]
In https://github.com/ltsp/community/issues/14#issuecomment-529095186 you said:

> no files, settings saved at logout (e.g an user who changes desktop);

In this case you **don't** need multiple users. You only need one user "guest", and multiple home directories on your server:

On pc01:
FSTAB_NFS="server:/home/pc01 /home nfs defaults,nolock 0 0"

On pc02:
FSTAB_NFS="server:/home/pc02 /home nfs defaults,nolock 0 0"

Etc etc. This way the guest will be able to always use the same /home/guest path, so that you can have one template for all users.

To prevent the users from modifying settings, read https://wiki.gnome.org/Projects/dconf/SystemAdministrators; it's not related to LTSP.

**Comment by gfreccia** [comment]
Hi Alkis,

I see on lts.conf manual:
_AUTOLOGIN="pc/guest" means "automatically log in as guest01 in pc01, as guest02 in pc02 etc"._

I've assigned on my DHCP server the hostname _pc01_ at the client PC.
I've set _AUTOLOGIN="pc/guest"_ on my lts.conf.
Now I see a "guest" user together the other users at login page, but no autologin working and I don't know which password to put!

**Comment by alkisg** [comment]
You only need one user, not many.
Create a user with username=guest.
In ltsp.conf, under [client], set:

- AUTOLOGIN="guest"
- RELOGIN=1
- PASSWORDS_GUEST="guest/"
- LTSPDM_USERS="guest"

This will autologin the guest user, without any password as you're using NFS and it's not needed, and if you log out, it will relogin if you're using GDM or SDDM (you're not, you're using lightdm), or it will show the guest user only at the login screen, and allow the guest to login with just pressing Enter.

**Comment by gfreccia** [comment]
Hi Alkis,

I've set as suggested and created an user "guest" (password "guest") using _adduser_ command.
I also tried with _AUTOLOGIN="pc/guest"_.
I see _guest_ user on login window but Autologin doesn't work (nor guest login with "guest" as password).

**Comment by alkisg** [comment]
Can you upload your ltsp.conf?

Btw are you running `ltsp initrd` after editing ltsp.conf?

**Comment by gfreccia** [comment]
Here!
[ltsp.txt](https://github.com/ltsp/community/files/3663275/ltsp.txt)

**Comment by gfreccia** [comment]
Btw are you running ltsp initrd after editing ltsp.conf?
Yes!

**Comment by alkisg** [comment]
When the client boots, can you login with guest and *enter*, i.e. without entering a password?

**Comment by gfreccia** [comment]
No!

**Comment by alkisg** [comment]
OK I think something is wrong in your setup, not in ltsp.conf or in ltsp itself.

It might be best to come to [IRC](http://ltsp.org/irc) for live help instead of here.
